PR for #2867: Translate Order Mode In OrderByDropdown

Robin-Smets opened the pull request:
- implemented method to translate ordermode in arrow unicode symbols and added css class to ensure proper display

abarz722 commented:
It would be nice, if also the new compiler warnings could be omitted, e.g.:
OrderByDropdown.razor(132,102): warning CS8602: Dereferenzierung eines möglichen Nullverweises.

Robin-Smets commented:
> It would be nice, if also the new compiler warnings could be omitted, e.g.: OrderByDropdown.razor(132,102): warning CS8602: Dereferenzierung eines möglichen Nullverweises.

I would be happy to do that and suggest that we postpone the merge until then. @tpurschke

tpurschke commented on roles/ui/files/FWO.UI/Shared/OrderByDropdown.razor:
```diff
@@ -178,4 +180,25 @@
+    {
+        switch (orderMode)
+        {
+            case (OrderMode.Asc):
```
i guess special characters should be avoided here - can't we use something from oi?

Robin-Smets replied:
I changed that in respect of our talk during the meeting to open iconic icons.

Robin-Smets commented:
> It would be nice, if also the new compiler warnings could be omitted, e.g.: OrderByDropdown.razor(132,102): warning CS8602: Dereferenzierung eines möglichen Nullverweises.

I refactored the code in a way that there are no warnings.
 
Additionally I found a bug and solved it. Now the Collection is properly ordered when the dialog pops up.

Robin-Smets commented:
> in the summary of method CompareSubnetMasks - could you please add possible input strings that are correctly compared? copilot suggests:
> 
> ```
>     /// Compares two subnet masks and returns an integer that indicates their relative order.
>     /// Recognized subnet formats include:
>     /// - CIDR notation (e.g., "24")
>     /// - IP address format (e.g., "255.255.255.0")
>     /// - Empty string (treated as the smallest possible subnet)
>     /// The comparison follows these rules:
>     /// - Empty strings are considered smaller than any other subnet mask.
>     /// - CIDR notation is compared numerically.
>     /// - IP addresses are compared by their numeric value.
>     /// - If one subnet mask is in IP format and the other is not, the IP format is considered smaller.
>     /// </summary>
>     /// <param name="subnetMask1">The first subnet mask to compare.</param>
>     /// <param name="subnetMask2">The second subnet mask to compare.</param>
>     /// <returns>
>     /// A signed integer that indicates the relative values of the subnet masks:
>     /// - Less than zero: <paramref name="subnetMask1"/> is less than <paramref name="subnetMask2"/>.
>     /// - Zero: <paramref name="subnetMask1"/> is equal to <paramref name="subnetMask2"/>.
>     /// - Greater than zero: <paramref name="subnetMask1"/> is greater than <paramref name="subnetMask2"/>.
>     /// </returns>
> ```

Like we discussed in the call I deleted the unused method alltogether.

Furthermore I fixed the bug that you experienced. I tested it and everthing works fine.
